### PR TITLE
mavfft_isb: add support for calculation of harmonic notch parameters

### PR DESCRIPTION
**Pull request by andyp1per**
This allows mavfft_isb to be invoked with the argument --notch-params and it will guess values for the harmonic notch based on the discovered peaks:
```
MacBook-Air:pymavlink andy$ ./tools/mavfft_isb.py --notch-params ~/Dropbox/Public/logs/Pre_Filter_User.bin 
Processing log /Users/andy/Dropbox/Public/logs/Pre_Filter_User.bin
.......................................................
55604s messages  74329 messages/second  4490477 kB/second
Extracted 208 fft data sets
Throttle average 0.227049
Sensor: Gyro[0]
Sensor: Accel[0]
Peaks: [ 39.01628017  81.93418837 120.95046854 159.96674871 241.90093708]
INS_HNTCH_REF = 0.2270
INS_HNTCH_FREQ = 39.0
INS_HNTCH_BW = 19.5
```

### CODE REVIEW

**Comment by andyp1per** [comment]
Asked for feedback on https://discuss.ardupilot.org/t/tuning-the-harmonic-notch/48836/631

**Review by amilcarlucas — COMMENTED** [review]
Nice, simple one

**Comment by peterbarker** [comment]
On Tue, 8 Sep 2020, Andy Piper wrote:

> MacBook-Air:pymavlink andy$ ./tools/mavfft_isb.py --notch-params ~/Dropbox/Public/logs/Pre_Filter_User.bin 
> Processing log /Users/andy/Dropbox/Public/logs/Pre_Filter_User.bin
> .......................................................
> 55604s messages  74329 messages/second  4490477 kB/second
> Extracted 208 fft data sets
> Throttle average 0.227049
> Sensor: Gyro[0]
> Sensor: Accel[0]
> Peaks: [ 39.01628017  81.93418837 120.95046854 159.96674871 241.90093708]
> INS_HNTCH_REF = 0.2270
> INS_HNTCH_FREQ = 39.0
> INS_HNTCH_BW = 19.5

This is starting to get very tightly coupled with ArduPilot version.

Perhaps we should consider moving this into the ArduPilot Tools directory?

File: tools/mavfft_isb.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
 import copy
 
 from argparse import ArgumentParser
+import scipy.signal as signal
```
Comment (peterbarker) [inline tools/mavfft_isb.py:16]: I'm really not a fan of importing symbols in like this.

**Comment by peterbarker** [comment]
Merged, thanks!

**Pull request merged**